src/main/java/appeng/util/helpers/P2PHelper.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2015, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.util.helpers;
+
+
+import com.google.common.base.Preconditions;
+
+import appeng.api.util.AEColor;
+
+
+public class P2PHelper
+{
+
+	public AEColor[] toColors( short frequency )
+	{
+		final AEColor[] colours = new AEColor[4];
+
+		for( int i = 0; i < 4; i++ )
+		{
+			int nibble = ( frequency >> 4 * ( 3 - i ) ) & 0xF;
+
+			colours[i] = AEColor.values()[nibble];
+		}
+
+		return colours;
+	}
+
+	public short fromColors( AEColor[] colors )
+	{
+		Preconditions.checkArgument( colors.length == 4 );
+
+		int t = 0;
+
+		for( int i = 0; i < 4; i++ )
+		{
+			int code = colors[3 - i].ordinal() << 4 * i;
+
+			t |= code;
+		}
+
+		return (short) ( t & 0xFFFF );
```

Comment:
`AEColor.TRANSPARENT` is not a color. It is the absence of any color (or any depending on the use case). The only reason it is part of the enum, is that wrapping everything into a null check and then having to fall back to default values is tedious. Similar to why we use `AEPartLocation` and not `EnumFacing`.

Sure I could safeguard it, but it is never intended to be used externally. Or probably not even internally. The sole reason I added it, is because I can test the `toColors` more easily.